faker/providers/color/color.py

```
@@ -0,0 +1,276 @@
+
+    def pick_brightness(self, h, s, luminosity):
+        b_min = self.get_minimum_brightness(h, s)
+        b_max = 100
+
+        if luminosity == 'dark':
+            b_max = b_min + 20
+        elif luminosity == 'light':
+            b_min = (b_max + b_min) / 2
+        elif luminosity == 'random':
+            b_min = 0
+            b_max = 100
+
+        return self.random_within([b_min, b_max])
+
+    def set_format(self, hsv, color_format):
+        if color_format == 'hsv':
+            color = 'hsv({}, {}, {})'.format(*hsv)
+
+        elif color_format == 'hsl':
+            hsl = self.hsv_to_hsl(hsv)
+            color = 'hsl({}, {}, {})'.format(*hsl)
+
+        elif color_format == 'rgb':
+            rgb = self.hsv_to_rgb(hsv)
+            color = 'rgb({}, {}, {})'.format(*rgb)
+
+        else:
+            rgb = self.hsv_to_rgb(hsv)
+            color = '#{:02x}{:02x}{:02x}'.format(*rgb)
+
+        return color
+
+    def get_minimum_brightness(self, h, s):
+        lower_bounds = self.get_color_info(h)['lower_bounds']
+
+        for i in range(len(lower_bounds) - 1):
+            s1 = lower_bounds[i][0]
+            v1 = lower_bounds[i][1]
+
+            s2 = lower_bounds[i + 1][0]
+            v2 = lower_bounds[i + 1][1]
+
+            if s1 <= s <= s2:
+                m = (v2 - v1) / (s2 - s1)
+                b = v1 - m * s1
+
+                return m * s + b
+
+        return 0
+
+    def get_hue_range(self, color_input):
+        if isinstance(color_input, (int, float)) and 0 <= color_input <= 360:
+            color_input = int(color_input)
+            return [color_input, color_input]
+
+        elif isinstance(color_input, six.string_types) and color_input in self.colormap:
+            return self.colormap[color_input]['hue_range']
+
+        try:
+            v1, v2 = color_input
+            v1 = int(v1)
+            v2 = int(v2)
+        except (ValueError, TypeError):
+            return [0, 360]
```

I'm not sure I'd want it to silently fallback to defaults. I'd rather have it throw an exception if the user is passing invalid values.

---

Alright, I will change this behavior.